### DISCUSSION TITLE
Would it be possible to add EN 13757 (M-Bus) protocol support?

### DISCUSSION THREAD

**Discussion by ssuukk** [jgromes/RadioLib#954] [Ideas]
This is a standard european smart meter protocol:

https://en.wikipedia.org/wiki/Meter-Bus
https://m-bus.com/
https://joinup.ec.europa.eu/collection/ict-standards-procurement/solution/en-13757-12014-communication-systems-meters-and-remote-reading-meters-part-1-data-exchange/about

**Comment by jgromes** [comment]
> Would it be possible
 
 Most likely yes. Is it wort it? That's the better question. Why should this protocol be added to RadioLib?

**Comment by ssuukk** [comment]
Well, being European standard for home water/gas/energy counters it makes
using them much cheaper in home automation solutions (like Home Assistant
with ESPHome running RadioLib) than specialized zigbee/thread/wifi/whatever
attachments (i.e. three phase zigbee power sensor is about 100 euro, not to
mention a fee for a professional electrician, while M-Bus is there, ready
for use...)

pon., 5 lut 2024, 21:50 użytkownik Jan Gromeš ***@***.***>
napisał:

**Reply by jgromes** [reply]
> makes using them much cheaper in home automation solutions

I don't have any information to judge that on, I did not have to install either system and so cannot compare. Though I will note that the actual standard is (as many ISO/EN standards are, for some reason) hidden behind a multi-hundred EUR paywall. Not exactly cheap ...

What exactly is the use case here? Do you want to create your own "smart meters"? Do you want to create the gateways? Are you willing to accept some limitations (e.g. not being able to compyl fully with the standard, due to not supporting some modulations)? ... And even if RadioLib was to implement this protocol, how would we test it?

**Comment by ssuukk** [comment]
I have such smart meters in my apartment block, since I'm using already
using radio lub for other home automation devs, I just thought it could to
this one more thing.

wt., 6 lut 2024, 21:12 użytkownik Jan Gromeš ***@***.***>
napisał:

**Reply by jgromes** [reply]
Well ... nobody is stopping you from using RadioLib as the RF driver and then implementing the protocol yourself ;) At the end of the day, even if the Mbus protocol was added to RadioLib, the implementation would just be calling some `transmit()` and `receive()` methods, that's it. So it's comparably difficult for both of us (with you actually having the advantage of being able to test).

Unless more people request this, I cannot justify working on this in my own time. Pull request might get accepted though.
